### PR TITLE
feat: CDW-54 add options for onboarding form

### PR DESCRIPTION
**Pull request by ifasama**
Type of change

[] New feature

[x] Bug fix

[] Documentation

[] Code refactor/tech debt

[] Infrastructure

Description

Changing onboarding forms (mentee and mentor) for job title, experience, and interest questions to predetermined options, integrated with previously created forms.

The code works as intended, but hoping for reviews on whether it can be better coded. Like on the array part in the constant or how to integrate the select. Thanks

### CODE REVIEW

File: frontend/src/constants.tsx
Comment (timchang87) [inline frontend/src/constants.tsx:1]: The constants in this file were designed to sort of drives FormRow and to help render the appropriate Chakra component. There is an opportunity to optimize the data here.

If you look at the constants I created like MENTEE_FIELDS and MENTOR_FIELDS, you'll see what the structure of the data looks like for textArea and inputs. You can do something similar for select fields that might be specific to mentor or mentees. If they belong in both you can use QUESTIONNAIRE_FIELDS.

Also, if you look at your constants, there's a lot of overlap of properties (i.e. there are 6 objects that have name called experience). There may be an opportunity to bundle up properties here.

Changes here will most likely mean changes in your other components.

File: frontend/src/components/FormRow.tsx
Comment (timchang87) [inline frontend/src/components/FormRow.tsx:1]: FormRow was designed to be wrapper component, to render any Charka component dynamically depending on the prop. This was designed to broadly cover various use cases in regards to forms.

With Select, this wrapper component can now render 3 different form components. Although ternary and nested ternary will work, we might be able to make this more readable using a switch statement. I am not sure if this is best practice but the reasoning is that we need logic to determine which form component to render. And if we are going to take into account other form components in the future, using a switch might be a maintainable and readable approach.

**Review by timchang87 — CHANGES_REQUESTED** [review]

File: frontend/src/constants.tsx
Comment (ifasama) [reply frontend/src/constants.tsx:1]: Yeah, I understand. I tried to put the values inside QUESTIONNAIRE_FIELDS before but it did not work thus I tried this way. I am fully aware it is not optimal, let me try again.

File: frontend/src/components/FormRow.tsx
Comment (ifasama) [reply frontend/src/components/FormRow.tsx:1]: I think ternary for three options is okay, no? Maybe if we'll have more options then we can do switch statement.

Comment (timchang87) [reply frontend/src/components/FormRow.tsx:1]: This is definitely a preference and may be different from person to person. But to me I think we can improve readability and maintainability now that an additional component will be rendered here and since ternary conditions are used for boolean situations. Again, not sure if using `switch` is best practice but I think it will at least help with readability and maintainability.

Comment (jtabba) [reply frontend/src/components/FormRow.tsx:1]: No nested ternaries in the JSX! ever

File: frontend/src/components/FormOption.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        LISTS
+        .filter(list => list.name === name) 
+        .map((list, idx) => 
+            <option key={idx} value={list.values}>{list.values}</option>)
+        }
+        </>
+)
+}
+
+export default FormOption;
```
Comment (jtabba) [inline frontend/src/components/FormOption.tsx:21]: Install Prettier if you don't have it so your code is auto formatted

File: frontend/src/components/FormRow.tsx [hunk 2]
```diff
@@ -21,14 +22,15 @@ const FormRow: FC<FormRowProps> = ({
...
   return (
     <Flex flexDirection={flexDirection}>
       <FormLabel fontSize={fontSize}>{label}</FormLabel>
       {inputType === 'textArea' ? (
         <Textarea name={name} value={value} onChange={onChange} />
-      ) : (
+      ) : inputType === 'input' ? (
         <Input name={name} value={value} onChange={onChange}></Input>
-      )}
+      ): (<Select placeholder="Select one option" name={name} onChange={onChange}><FormOption name={name}/></Select>)}
```
Comment (jtabba) [inline frontend/src/components/FormRow.tsx]: Turn this into an if or switch statement

**Review by jtabba — CHANGES_REQUESTED** [review]
Made some minor comments
